src/core/vectortile/qgsvectortilestructure.cpp

```diff
@@ -0,0 +1,90 @@
+/***************************************************************************
+  qgsvectortilestructure.cpp
+  --------------------------------------
+  Date                 : March 2022
+  Copyright            : (C) 2022 by Nyall Dawson
+  Email                : nyall dot dawson at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsvectortilestructure.h"
+#include "qgstiles.h"
+#include "qgsvectortileutils.h"
+#include "qgsarcgisrestutils.h"
+
+QgsVectorTileStructure QgsVectorTileStructure::fromWebMercator()
```

Comment:
I believe with custom CRS the approach with bounds() is not going to be great. In some countries they have kind of standard tile matrix set for their national CRS, but I doubt it would be the same as the bounds() most of the time (just because of the fact that you want tile matrix boundary to be a square or built by multiple squares). It would be more appropriate to have a collection of "well known tile matrix sets" mainly for countries where there's such agreement (I don't know of any authority for this, but I have seen that MapTiler have a list of such tile matrix sets for some European countries).